protobuf/v1/snapshot.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+
+import "common.proto";
+package mayastor.v1;
+
+/// Represenation of a single snapshot. A list this object is to
+/// be returned as part of list snapshots request.
+message Snapshot {
+    string snapshot_uuid = 1; // uuid of the snapshot.
+    string snapshot_name = 2; // name of the snapshot.
+    uint64 snapshot_size = 3; // amount of bytes referenced by snapshot.
+    uint64 num_clones = 4; // number of clones created from this snapshot.
+    google.protobuf.Timestamp timestamp = 5; // snapshot creation_timestamp.
+    string replica_uuid = 6; // uuid of the replica this snapshot is taken from.
```

Comment:
this has replica uuid which is not common?

Reply:
> this has replica uuid which is not common?

I think this can be renamed to `parent_uuid`. This field indicate the source uuid from which the snapshot is created. It can be `replica_uuid` or `nexus_uuid` based on flow it handle.

Reply:
I'm not sure, nexus does not return this information at all, so this snapshot seems to be useful only for a replica.

Reply:
The way I interpreted this filed is: Irrespective of nexus snapshot or replica snapshot, a snapshot eventually is from a replica underneath. So that replica id is to be set here.

Reply:
But then this is not a snapshot, it's a ReplicaSnapshot!

Reply:
Yes, at the most granular level it is a replica snapshot, a vector of which need to be returned for list query by nexus or replica or blank. I think we have two options - 
1. Let it be called as ReplicaSnapshot and replica_uuid can contain inside it. List by nexus, replica, or blank all should be ok with this.
2. Rename as Snapshot, remove replica_uuid from inside this. In this case I think we'll have to provide a way to group these Snapshot objects in list results. 

First one is less work. What do you think?

Reply:
The nexus isn't going to return this snapshot AFAIK,  nexus has no visibility into the actual replica snapshots themselves. The CP will have to list snapshots from replicas api to find out all this information. So I think it should remain ReplicaSnapshot and have replica_uuid within it, and remain in the replica protobuf.

Reply:
hmm, I'm kinda confused now what 'list by nexus' is going to return. Presumably, CP will break 'list by nexus' into multiple 'list by replica' calls and send the rpc to DP. OR 'list by nexus' will return the top level  snapshot ids(what use they'll be). I'll better hold this off till we have a discussion sometime next week to come on same page.